modules/openstack/openstack.go

```diff
@@ -47,12 +47,16 @@ func NewOpenStack(
 	log logr.Logger,
 	cfg AuthOpts,
 ) (*OpenStack, error) {
+
 	opts := gophercloud.AuthOptions{
 		IdentityEndpoint: cfg.AuthURL,
 		Username:         cfg.Username,
 		Password:         cfg.Password,
 		TenantName:       cfg.TenantName,
 		DomainName:       cfg.DomainName,
+		Scope: &gophercloud.AuthScope{
+			System: true,
```

Comment:
Shouldn't this be based on some condition or do we always want this client with system scope?

Reply:
Something like this;
if os.Getenv("OS_SYSTEM_SCOPE") != "all"
     System: false

Reply:
As you said, I am not sure if we need it, but you could add a Scope pointer to the above `AuthOpts` and if it is not nil you can set it in `NewOpenStack`. with this there would be no need to the keystone-operator